components/camel-milo/src/main/java/org/apache/camel/component/milo/client/MiloClientProducer.java

```diff
@@ -53,7 +57,15 @@ public boolean process(Exchange exchange, AsyncCallback async) {
 
         final CompletableFuture<?> future;
 
-        if (this.methodId == null) {
+        if (msg.getHeaders().containsKey(HEADER_NODE_IDS)) {
```

Comment:
You're changing the default behavior here, so it would be better if you add more documentation under src/main/docs of the project

Reply:
done.